### PR TITLE
Add file based password authenticator plugin

### PR DESCRIPTION
**Pull request by electrum**
Supersedes https://github.com/prestosql/presto/pull/797

### CODE REVIEW

File: presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            return BCrypt.Version.VERSION_2A.parser.parse(password.getBytes(UTF_8)).cost;
+        }
+        catch (IllegalBCryptFormatException e) {
+            throw new HashedPasswordException("Invalid BCrypt password", e);
+        }
+    }
+
+    public static int getPBKDF2Iterations(String password)
+    {
+        Iterator<String> parts = Splitter.on(":").split(password).iterator();
```
Comment (dain) [inline presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java]: Limit 2?

File: presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    public static boolean isPBKDF2PasswordValid(String inputPassword, String hashedPassword)
+    {
+        List<String> parts = Splitter.on(":").splitToList(hashedPassword);
+        if (parts.size() != 3) {
+            throw new HashedPasswordException("Invalid PBKDF2 Password");
+        }
+
+        int iterations = Integer.parseInt(parts.get(0));
```
Comment (dain) [inline presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java]: catch number format exception?

File: presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        byte[] salt = base16().lowerCase().decode(parts.get(1));
+        byte[] hash = base16().lowerCase().decode(parts.get(2));
```
Comment (dain) [inline presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java]: catch decode exception?

File: presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java (hunk 1 above)
Comment (electrum) [reply presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java]: It actually has 3 parts. I'll change it to be like `isPBKDF2PasswordValid` and validate it has exactly that many.

**Pull request closed without merging**